mina-transaction/transaction-leaves.ts

```diff
@@ -70,3 +71,30 @@ const TransactionVersion = {
   ...provable(UInt32),
   empty: () => UInt32.from(protocolVersions.txnVersion),
 };
+
+type MayUseToken = {
+  parentsOwnToken: Bool;
+  inheritFromParent: Bool;
+};
+const MayUseToken = {
+  ...provable({
+    parentsOwnToken: Bool,
+    inheritFromParent: Bool,
+  }),
+  check: ({ parentsOwnToken, inheritFromParent }: MayUseToken) => {
+    if (
+      !(parentsOwnToken instanceof Bool) ||
+      !(inheritFromParent instanceof Bool)
+    ) {
+      throw Error(
+        'MayUseToken: parentsOwnToken and inheritFromParent must be Bool'
+      );
+    }
```

Comment:
I meant to add in-circuit checks instead:
```ts
Bool.check(parentsOwnToken);
Bool.check(inheritFromParent);
```

Reply:
Gotcha :sweat_smile: